headers.go

```diff
@@ -62,6 +62,19 @@ func (h ProtectedHeader) SetAlgorithm(alg Algorithm) {
 	h[HeaderLabelAlgorithm] = int(alg)
 }
 
+// GetAlgorithm gets the algorithm value from the algorithm header.
+func (h ProtectedHeader) GetAlgorithm() (Algorithm, error) {

```

Comment:
Go tends to avoid prefixing getters with `Get`: https://go.dev/doc/effective_go#Getters. `Algorithm()` feels more idiomatic,